.github/workflows/native-unix.yml

```diff
@@ -178,28 +173,134 @@ jobs:
+          persist-credentials: false
+      - name: Get Date
+        id: get-date
+        shell: bash
+        run: |
+          echo "today=$(/bin/date -u '+%Y%m%d')" >> $GITHUB_OUTPUT
+      - name: Cache Conda
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/conda_pkgs_dir
+          key: conda-${{ runner.os }}-${{ steps.get-date.outputs.today }}-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/**') }}
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          use-only-tar-bz2: false
+          use-mamba: true
+      - name: Install Dependencies
+        shell: bash -l {0}
+        run: |
+          mamba install -c conda-forge \
+            --file ci/conda_env_cpp.txt
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.6
+          check-latest: true
+          cache: true
+          cache-dependency-path: go/adbc/go.sum
       - name: Build PostgreSQL Driver
         shell: bash -l {0}
         run: |
           env BUILD_ALL=0 BUILD_DRIVER_POSTGRESQL=1 ./ci/scripts/cpp_build.sh "$(pwd)" "$(pwd)/build"
       - name: Test PostgreSQL Driver
         shell: bash -l {0}
-        if: matrix.os == 'ubuntu-latest'
         env:
           BUILD_ALL: "0"
           BUILD_DRIVER_POSTGRESQL: "1"
           ADBC_POSTGRESQL_TEST_URI: "postgres://localhost:5432/postgres?user=postgres&password=password"
         run: |
           ./ci/scripts/cpp_test.sh "$(pwd)" "$(pwd)/build"
-      - name: Build Driver Manager
+      - name: Login To Dremio
         shell: bash -l {0}
+        id: bootstrap
         run: |
-          env BUILD_ALL=0 BUILD_DRIVER_MANAGER=1 ./ci/scripts/cpp_build.sh "$(pwd)" "$(pwd)/build"
-      - name: Test Driver Manager
+          LOGIN=`curl -X POST -H "Content-Type: application/json" "http://localhost:9047/apiv2/login" \
+            -d '{"userName":"dremio","password":"dremio123"}'`
+          echo "login_resp=$LOGIN" >> $GITHUB_OUTPUT
+      - name: Auth Token
         shell: bash -l {0}
+        id: auth
         run: |
-          env BUILD_ALL=0 BUILD_DRIVER_MANAGER=1 ./ci/scripts/cpp_build.sh "$(pwd)" "$(pwd)/build"
-          env BUILD_ALL=0 BUILD_DRIVER_MANAGER=1 ./ci/scripts/cpp_test.sh "$(pwd)" "$(pwd)/build"
+          echo "token=_dremio${{ fromJSON(steps.bootstrap.outputs.login_resp).token }}" >> $GITHUB_OUTPUT
+      - name: Add Samples Catalog
+        shell: bash -l {0}
+        run: |
+          curl -X PUT -H "content-type: application/json" -H "authorization: ${{ steps.auth.outputs.token }}" \
+            "http://localhost:9047/apiv2/source/Samples/" -d '{"config":{"externalBucketList":["samples.dremio.com"],
+            "credentialType":"NONE","secure":false,"propertyList":[]},"name":"Samples","accelerationRefreshPeriod":3600000,
+            "accelerationGracePeriod":10800000,"accelerationNeverRefresh":true,"accelerationNeverExpire":true,"type":"S3"}'
```

Comment:
Is there a way to put up local sample data? It does seem a little rude to hit their bucket from CI

Reply:
(Maybe as a follow up to this)

Reply:
Well I could add the postgresql that we spun up as a source lol, that'd work